glean-core/android/src/main/java/mozilla/telemetry/glean/private/EventMetricType.kt

```diff
@@ -64,45 +98,36 @@ class EventMetricType<ExtraKeysEnum : Enum<ExtraKeysEnum>>(
      *              The maximum length for values is defined by [MAX_LENGTH_EXTRA_KEY_VALUE]
      */
     fun record(extra: Map<ExtraKeysEnum, String>? = null) {
-        // if (!shouldRecord(logger)) {
-        //     return
-        // }
-
-        // // We capture the event time now, since we don't know when the async code below
-        // // might get executed.
-        // val monotonicElapsed = SystemClock.elapsedRealtime()
-
-        // // Convert the extra key enums to strings before passing to the storage engine
-        // val extraStrings = extra?.let { extra ->
-        //     // There are two extra "keys" in play here:
-        //     //   1. The Kotlin enumeration names, in CamelCase
-        //     //   2. The keys sent in the ping, in snake_case
-        //     // Here we need to get (2) to send in the ping.
-        //     if (extra.size > 0) {
-        //         val result = mutableMapOf<String, String>()
-        //         for ((k, v) in extra) {
-        //             allowedExtraKeys.getOrNull(k.ordinal)?.let { stringKey ->
-        //                 result[stringKey] = v
-        //             } ?: run {
-        //                 logger.debug("No string value for enum ${k.ordinal}")
-        //             }
-        //         }
-        //         result
-        //     } else {
-        //         null
-        //     }
-        // }
-
-        // @Suppress("EXPERIMENTAL_API_USAGE")
-        // Dispatchers.API.launch {
-        //     // Delegate storing the event to the storage engine.
-        //     EventsStorageEngine.record(
-        //         metricData = this@EventMetricType,
-        //         monotonicElapsedMs = monotonicElapsed,
-        //         extra = extraStrings
-        //     )
-        // }
-        Log.e(LOG_TAG, "EventMetricType.record is a stub")
+        if (!shouldRecord()) {
+            return
+        }
+
+        // We capture the event time now, since we don't know when the async code below
+        // might get executed.
+        val timestamp = SystemClock.elapsedRealtime()
+
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            // The Map is sent over FFI as a pair of parallel Arrays
```

Comment:
What are parallel arrays?

Reply:
Sure -- I can clarify in the comment.  [Parallel arrays](https://en.wikipedia.org/wiki/Parallel_array) are basically records where the fields have been separated out into separate arrays.